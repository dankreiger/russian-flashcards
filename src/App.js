import React, { Component } from 'react';
import Flashcard from './Flashcard';
import vocabulary from './vocabulary';
import './App.css';

class App extends Component {
  constructor() {
    super();
    this.state = { flipped: false, card: 0 };
  }

  flipCard = () => {
    this.setState({flipped: !this.state.flipped});
  }

  nextCard = () => {
    this.setState({flipped: false, card: this.state.card === vocabulary.length - 1 ? 0 : this.state.card + 1});
  }

  render() {
    const { card, flipped } = this.state;

    return (
      <div className="App">
        <Flashcard card={card} flipped={flipped} />
        <div className="flex-buttons">
          <button className="btn blue" onClick={this.flipCard}>
            <span>Flip</span>
          </button>
          <button className="btn" onClick={this.nextCard}>
            <span>Next</span>
          </button>
        </div>
      </div>
    );
  }
}

export default App;
