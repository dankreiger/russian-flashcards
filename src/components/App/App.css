.App {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  height: 100vh;
  -webkit-box-pack: space-around;
      -ms-flex-pack: space-around;
          justify-content: space-around;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
}

.starting-language-menu {
  -ms-flex-item-align: flex-end;
    align-self: flex-end;
}

select {
  margin: 0;
  font-family: inherit;
  font-size: inherit;
  line-height: inherit;  
}
