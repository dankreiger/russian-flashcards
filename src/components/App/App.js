import React, { Component } from 'react';
import Flashcard from '../Flashcard/Flashcard';
import Button from '../Button/Button';
import vocabulary from '../../static/vocabulary';
import Artyom from 'artyom.js';
import './App.css';

class App extends Component {
  constructor() {
    super();
    this.state = { flipped: false, card: 0, currentLang: "Russian" };
  }

  flipCard = () => {
    this.setState({flipped: !this.state.flipped});
  }

  nextCard = () => {
    this.setState({flipped: false, card: this.state.card === vocabulary.length - 1 ? 0 : this.state.card + 1});
  }

  sayWord = () => {
    let artyom = new Artyom();
    artyom.say(vocabulary[this.state.card].word, {
      lang:"ru-RU"
    });
  }

  render() {
    const { card, flipped, currentLang } = this.state;

    return (
      <div className="App">
        <div className="flex-buttons-column">
          <Button classes="btn blue sayWord" command={this.sayWord} text="Say"/>
          <Button classes="btn" text={currentLang} />
        </div>
        <Flashcard card={card} flipCard={this.flipCard} flipped={flipped} />
        <div className="flex-buttons">
          <Button classes="btn orange flipCard" command={this.flipCard} text="Flip"/>
          <Button classes="btn nextCard" command={this.nextCard} text="Next"/>
        </div>
        {/* <Buttons sayWord={this.sayWord} flipCard={this.flipCard} nextCard={this.nextCard}/> */}
      </div>
    );
  }
}

export default App;
