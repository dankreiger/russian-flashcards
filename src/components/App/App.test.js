import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';


import App from './App';

describe('<App />', () => {
  let app;
  beforeEach(() => {
    app = mount(<App />);
  });

  it('renders a <Flashcard /> component', () => {
    expect(app.find('Flashcard').exists()).toBe(true);
  });

  it('renders correctly', () => {
    expect(toJson(app)).toMatchSnapshot();
  });

  it('renders the first card unflipped with the correct text', () => {
    expect(app.state().flipped).toBe(false);
    expect(app.state().card).toEqual(0);

    expect(app.find('.russian').hasClass('hidden')).toBe(false);
    expect(app.find('.english').hasClass('hidden')).toBe(true);

    expect(app.find('.russian').find('.lead').text()).toEqual('Да');
    expect(app.find('.russian').find('.phonetic').text()).toEqual('[dá]');

  })

  it('flips the flashcard on button click', () => {
    app.find('.flipCard').simulate('click');

    expect(app.find('.Flashcard').hasClass('flipped')).toBe(true);

    expect(app.state().flipped).toBe(true);
    expect(app.state().card).toEqual(0);


    expect(app.find('.russian').hasClass('hidden')).toBe(true);
    expect(app.find('.english').hasClass('hidden')).toBe(false);

    expect(app.find('.english').find('.lead').text()).toEqual('Yes');
    expect(app.find('.english').find('.phonetic').text()).toEqual('[dá]');

  });

  it('goes to the next card on button click', () => {
    app.find('.nextCard').simulate('click');

    expect(app.find('.Flashcard').hasClass('flipped')).toBe(false);

    expect(app.state().flipped).toBe(false);
    expect(app.state().card).toEqual(1);


    expect(app.find('.russian').hasClass('hidden')).toBe(false);
    expect(app.find('.english').hasClass('hidden')).toBe(true);

    expect(app.find('.russian').find('.lead').text()).toEqual('Нет');
    expect(app.find('.russian').find('.phonetic').text()).toEqual("[n'ét]");

  });

  it('goes to the next card and flips after clicking buttons', () => {
    app.find('.nextCard').simulate('click');

    app.find('.flipCard').simulate('click');

    expect(app.find('.Flashcard').hasClass('flipped')).toBe(true);

    expect(app.state().flipped).toBe(true);
    expect(app.state().card).toEqual(1);


    expect(app.find('.russian').hasClass('hidden')).toBe(true);
    expect(app.find('.english').hasClass('hidden')).toBe(false);

    expect(app.find('.english').find('.lead').text()).toEqual('No');
    expect(app.find('.english').find('.phonetic').text()).toEqual("[n'ét]");

  });
});
