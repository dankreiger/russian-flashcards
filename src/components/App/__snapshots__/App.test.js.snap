// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders correctly 1`] = `
<App>
  <div
    className="App"
  >
    <Flashcard
      card={0}
      flipCard={[Function]}
      flipped={false}
    >
      <div
        className="Flashcard"
        onClick={[Function]}
      >
        <Russian
          card={
            Object {
              "phonetic": "[dá]",
              "translation": "Yes",
              "word": "Да",
            }
          }
          flipped={false}
        >
          <div
            className="russian"
          >
            <p
              className="lead"
            >
              Да
            </p>
            <p
              className="phonetic"
            >
              [dá]
            </p>
          </div>
        </Russian>
        <English
          card={
            Object {
              "phonetic": "[dá]",
              "translation": "Yes",
              "word": "Да",
            }
          }
          flipped={false}
        >
          <div
            className="english hidden"
          >
            <p
              className="lead"
            >
              Yes
            </p>
            <p
              className="phonetic"
            >
              [dá]
            </p>
          </div>
        </English>
      </div>
    </Flashcard>
    <Buttons
      flipCard={[Function]}
      nextCard={[Function]}
      sayWord={[Function]}
    >
      <div
        className="flex-buttons"
      >
        <button
          className="btn blue sayWord"
          onClick={[Function]}
        >
          <span>
            Say
          </span>
        </button>
        <button
          className="btn orange flipCard"
          onClick={[Function]}
        >
          <span>
            Flip
          </span>
        </button>
        <button
          className="btn nextCard"
          onClick={[Function]}
        >
          <span>
            Next
          </span>
        </button>
      </div>
    </Buttons>
  </div>
</App>
`;
